h/models/token.py

```diff
@@ -48,10 +58,13 @@ class Token(Base, mixins.Timestamps):
     #: A NULL value means it is a developer token.
     authclient = sqlalchemy.orm.relationship('AuthClient')
 
-    def __init__(self, **kwargs):
-        super(Token, self).__init__(**kwargs)
+    def __init__(self, expires=None, **kwargs):
```

Comment:
On second thought, I'm not a big fan of overriding SQLAlchemy model constructors too much, [we've done some recent work to remedy this in the `Group` model](https://github.com/hypothesis/h/commit/84ed007b2a151381299497789f015e087c3f31d9).
Can't you just change the `if` further down to: `if kwargs.get('expires')`?

In general I think we should try and leave the model constructors as much as possible, at least we should never mess with the signature. It's okay like in this case to do additional work (generating token, maybe generating a refresh token).